File: cpp/examples/octomap_registration.cpp
#include "registration_common.hpp"

int main() {
    auto set = examples::robot_description_fleet();
    set.datavizs.front().add_3d_octomap(
        "/map_3d_octomap_mesh",
        "map",
        {{"render_mode", std::string("surface_mesh")}, {"native_topic", std::string("/map_3d_octomap")}});
    return examples::register_or_fail(set);
}

File: rust/examples/octomap_registration.rs
mod common;

fn main() {
    let mut set = common::robot_description_fleet();
    set.datavizs[0].add_3d_octomap(
        "/map_3d_octomap_mesh",
        "map",
        Some(common::options(&[
            ("render_mode", serde_json::json!("surface_mesh")),
            ("native_topic", serde_json::json!("/map_3d_octomap")),
        ])),
    );
    common::register_or_fail(&mut set);
}
